File: LeetCode/0693. Binary Number with Alternating Bits/solution.go
/**
 * 204 / 204 test cases passed.
 * Runtime: 0 ms
 * Memory Usage: 1.8 MB 
 */
 func hasAlternatingBits(n int) bool {
    x := n ^ (n >> 1)    
    return (x & (x + 1)) == 0
}

File: LeetCode/0693. Binary Number with Alternating Bits/solution.cpp
/**
 * 204 / 204 test cases passed.
 * Runtime: 0 ms
 * Memory Usage: 5.7 MB 
 */
class Solution {
public:
    bool hasAlternatingBits(int n) {
        long x = n ^ (n >> 1);
        return (x & (x + 1)) == 0;
    }
};
